### PR TITLE
Export emulation settings even if some iterations failed

### PR DESCRIPTION
**Pull request by melutzky**

### CODE REVIEW

File: src/SDK/MonitoringApplianceEmulator/Models/EmulationRunSettings.cs [hunk 1]
```diff
@@ -112,19 +112,22 @@ public static EmulationRunSettings LoadEmulationRunSettings(string filePath)
...
                 Directory.CreateDirectory(emulatorRunSettingsPath);
 
-                // Creates the path of the file with the emulation run settings
-                string filePath = Path.Combine(emulatorRunSettingsPath, $"{this.StartTime:yyyy-MM-dd HH-mm-ss}_{this.EndTime:yyyy-MM-dd HH-mm-ss}_{this.AnalysisCadence.TimeSpan:hh'-'mm'-'ss}_{this.EmulationAlerts.Count}.json");
+                // Create file path with the emulation run settings
+                string filePath = (this.IterativeRunModeEnabled) ?
+                    Path.Combine(emulatorRunSettingsPath, $"{detectorName}_{this.StartTime:yyyy-MM-dd HH-mm-ss}_{this.EndTime:yyyy-MM-dd HH-mm-ss}_{this.AnalysisCadence.TimeSpan:hh'-'mm'-'ss}.json") :
+                    Path.Combine(emulatorRunSettingsPath, $"{detectorName}_{this.StartTime:yyyy-MM-dd HH-mm-ss}.json");
 
-                using (StreamWriter writer = File.CreateText(filePath))
```
Comment (bermanguy) [inline src/SDK/MonitoringApplianceEmulator/Models/EmulationRunSettings.cs]: >CreateText [](start = 50, length = 10)

why did you need to change it?

Comment (melutzky) [reply src/SDK/MonitoringApplianceEmulator/Models/EmulationRunSettings.cs]: I changed it back to how it was before.

---
In reply to: [238683625](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/32#discussion_r238683625) [](ancestors = 238683625,238670129)
